### PR TITLE
User memberships

### PR DESCRIPTION
**Pull request by cpackingham**
Resolves https://github.com/xanzy/go-gitlab/issues/944

- Added user memberships method
- Added a unit test with some test data

Tried to test this locally but I don't have access to an admin account. It looks like you might need to create a self-hosted gitlab instance, is that right?

### CODE REVIEW

File: testdata/get_user_memberships.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "source_type": "Project",
+    "access_level": "20"
+  },
+  {
+    "source_id": 3,
+    "source_name": "Group three",
+    "source_type": "Namespace",
+    "access_level": "20"
+  }
+]
```
Comment (svanharmelen) [inline testdata/get_user_memberships.json:14]: Please make sure the file ends with a newline.

File: users_test.go [hunk 2]
```diff
@@ -234,3 +236,23 @@ func TestActivateUser_UserNotFound(t *testing.T) {
...
+	path := fmt.Sprintf("/%susers/1/memberships", apiVersionPath)
+	mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		mustWriteHTTPResponse(t, w, "testdata/get_user_memberships.json")
+	})
+
+	opt := new(GetUserMembershipOptions)
+
+	memberships, _, err := client.Users.GetUserMemberships(1, opt)
+
```
Comment (svanharmelen) [inline users_test.go:253]:
```suggestion
```

File: users_test.go [hunk 3]
```diff
@@ -234,3 +236,23 @@ func TestActivateUser_UserNotFound(t *testing.T) {
...
+	require.NoError(t, err)
+
+	want := []*UserMembership{{SourceID: 1, SourceName: "Project one", SourceType: "Project", AccessLevel: "20"}, {SourceID: 3, SourceName: "Group three", SourceType: "Namespace", AccessLevel: "20"}}
+	require.Equal(t, want, memberships)
```
Comment (svanharmelen) [inline users_test.go:257]: While require is fine, assert would probably also be a good option here.

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Thanks @cpackingham! Few super minor comments, looks good otherwise 👍🏻

**Comment by cpackingham** [comment]
Okay I think I got all of the suggestions in, how does it look?

**Review by svanharmelen — APPROVED** [review]
Thanks @cpackingham! LGTM!

**Pull request merged**